lib/src/facts/external/execution_resolver.cc

```diff
@@ -29,38 +32,51 @@ namespace facter { namespace facts { namespace external {
 
         try
         {
-            string error;
-            each_line(
-                path,
-                [&](string const& line) {
+            bool resolved = false;
+            auto result = execute(path, 0,
+                {
+                    execution_options::trim_output,
+                    execution_options::merge_environment,
+                    execution_options::throw_on_failure,
+                    execution_options::convert_newlines
+                });
+
+            try {
+                auto node = YAML::Load(result.output);
+                for (auto const& kvp : node) {
+                    add_value(kvp.first.as<string>(), kvp.second, facts);
+
+                    // If YAML doesn't correctly parse, it will
+                    // sometimes just return an empty node instead of
+                    // erroring. Only claiming we've resolved if we
+                    // add at least one child value from the YAML
+                    // allows us to still pass on to the keyval
+                    // interpretation in those cases.
+                    resolved = true;
+                }
+            }  catch (YAML::Exception& ex) {
+                LOG_DEBUG("Could not parse executable fact output as YAML ({1})", ex.msg);
+            }
+
+            if (!resolved) {
+                std::vector<string> lines;
+                boost::split(lines, result.output, boost::is_any_of("\n"));
```

Comment:
@Iristyle The windows issue is this split. I could definitely just deal with the `\r` here, but it feels cleaner to me to let the I/O layer deal with that sort of platform-specific stuff. Maybe that just comes from me being used to `stdio` doing it, though.